### ISSUE TITLE
Parameter type for class method is incorrect when using ParamSpec

### ISSUE THREAD

**Issue by zhykzhykzhyk** [microsoft/pyright#4967]
**Describe the bug**
Parameter type for class method is replaced incorrectly when using ParamSpec

**To Reproduce**
Runing pyright with the code in the following section.

**Expected behavior**
Report argument x should be an int.
Report type of "A.f" is "(x: int, y: int, z: str) -> int"

**Screenshots or Code**
```python
# pyright: strict
from typing import Callable, ParamSpec, Self, reveal_type

P = ParamSpec('P')

class A:
    def __init__(self, x: int, y: int, z: str) -> None:
        self.a = x

    @classmethod
    def f(cls: Callable[P, Self], *args: P.args, **kwargs: P.kwargs) -> int:
        return cls(*args, **kwargs).a

a: int = A.f(x='123', y=12, z='23')
reveal_type(A.f)  # reporting (x: P.args, y: P.kwargs, z: str) -> int, which is incorrect
```

**VS Code extension or command-line**
pyright 1.1.303 (command-line)

**Additional context**
None

**Comment by zhykzhykzhyk** [comment]
found this still happens when f is not classmethod
```python
# pyright: strict
from typing import Callable, ParamSpec, Self, reveal_type

P = ParamSpec('P')

class A:
    def __call__(self, x: int, y: int, z: str) -> Self:
        self.a = x
        return self

    def f(self: Callable[P, Self], *args: P.args, **kwargs: P.kwargs) -> int:
        return self(*args, **kwargs).a

a: int = A().f(x='123', y=12, z='23')
reveal_type(A().f)  # reporting (x: P.args, y: P.kwargs, z: str) -> int, which is incorrect
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.304, which is now published. It will also be included in a future release of pylance.
